app/database/group_item_store_compute_all_access.go

```diff
@@ -102,11 +86,8 @@ func (s *GroupItemStore) computeAllAccess() {
 	mustNotBeError(err)
 	defer func() { mustNotBeError(stmtMarkExistingChildren.Close()) }()
 
-	// marking 'children' groups_items_propagate as 'done'
-	const queryMarkFinishedItems = `
-		UPDATE groups_items_propagate
-		SET propagate_access = 'done'
-		WHERE propagate_access = 'children'`
+	// deleting 'children' groups_items_propagate
+	const queryMarkFinishedItems = `DELETE FROM groups_items_propagate WHERE propagate_access = 'children'`
```

Comment:
Good.
So now you have delete, all those with "children", the one only remaining in `groups_items_propagate` are those with 'self'. So: 
* `// computation for groups_items marked as 'self' in groups_items_propagate.` -> `// computation for groups_items marked as 'self' in groups_items_propagate (so all of them).`
* `LEFT JOIN groups_items_propagate` -> remove
* `(groups_items_propagate.propagate_access = 'self' OR groups_items_propagate.id IS NULL) ` -> remove
* `WHERE groups_items_propagate.propagate_access = 'self'`  -> remove
* maybe I would move the `JOIN groups_items_propagate USING(id)` to the second line of the request, so that it is clearer that we work only the propagating ones.

Actually, for the  `LEFT JOIN groups_items_propagate` and its "where", it was quite useless. Do you find a reason?

Reply:
That would be true if we had a single-task system. But in AlgoreaBackend parellel requests can insert new rows with `propagate_access` = 'children' between the DELETE and UPDATE. And the UPDATE statement will see and process those new rows. So we still need to filter by `propagate_access` :(

Reply:
There is no lock anymore? 🤔 So it can be completely interleaved? How do you ensure that work correctly?
Ok, so you can take these comments, probably some still make a bit of sense.  And the left join is still strange to me.

Reply:
On the other hand, we should never commit a transaction with unprocessed `groups_items_propagate` rows...

Reply:
So should be okay unless someone inserts items via mysql console

Reply:
Fixed